src/Test/MakerTestEnvironment.php

```diff
@@ -0,0 +1,345 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Test;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\InputStream;
+
+/**
+ * @author Sadicov Vladimir <sadikoff@gmail.com>
+ */
+final class MakerTestEnvironment
+{
+    private $testDetails;
+
+    private $fs;
+
+    private $rootPath;
+    private $cachePath;
+    private $flexPath;
+
+    private $path;
+
+    private $snapshotFile;
+
+    /**
+     * @var MakerTestProcess
+     */
+    private $runnedMakerProcess;
+
+    private function __construct(MakerTestDetails $testDetails)
+    {
+        $this->testDetails = $testDetails;
+        $this->fs = new Filesystem();
+
+        $this->rootPath = realpath(__DIR__.'/../../');
+
+        $cachePath = $this->rootPath.'/tests/tmp/cache';
+
+        if (!$this->fs->exists($cachePath)) {
+            $this->fs->mkdir($cachePath);
+        }
+
+        $this->cachePath = realpath($cachePath);
+        $this->flexPath = $this->cachePath.DIRECTORY_SEPARATOR.'flex_project';
+
+        $this->path = $this->cachePath.DIRECTORY_SEPARATOR.$testDetails->getUniqueCacheDirectoryName();
+
+        $this->snapshotFile = $this->path.DIRECTORY_SEPARATOR.basename($this->path).'.json';
```

Comment:
I'm 99% sure there should be no problem with using `/` instead of `DIRECTORY_SEPARATOR`. In fact, I think using `/` is a Symfony standard.

Reply:
Agreed! Just some perfectionism :)